src/openms/include/OpenMS/ANALYSIS/TOPDOWN/PeakGroup.h

```diff
@@ -267,7 +273,7 @@ namespace OpenMS
     void sort();
 
     /// clear LogMzPeaks vectors to save memory.
```

Comment:
How about making a function that creates another PeakGroup but only with calculated results?

Reply:
hmm this could probably still mix two responsibilities (e.g., storing the calculated results and working with the raw data). Cleaner would probably be something that works on the raw data (e.g., accumulating raw peaks, filtering etc. ) and then, when done, creates a new PeakGroup object from it that contains calculated results. Not sure if that is feasible as I did not look deeply into the code.

Reply:
I see. The thing is sometimes it is needed to keep all the raw peaks (when write_detail == 1) for some users. And memory usage is an issue only when a really large dataset (e.g. TOF) is used. Hmm... I will try to think about a possible solution.